cas-client-support-saml/src/main/java/org/jasig/cas/client/validation/Saml11TicketValidator.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Licensed to Jasig under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Jasig licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.cas.client.validation;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.*;
+import org.jasig.cas.client.authentication.AttributePrincipalImpl;
+import org.jasig.cas.client.util.*;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Interval;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+import javax.xml.namespace.NamespaceContext;
+
+/**
+ * TicketValidator that can understand validating a SAML artifact.  This includes the SOAP request/response.
+ *
+ * @author Scott Battaglia
+ * @since 3.1
+ */
+public final class Saml11TicketValidator extends AbstractUrlBasedTicketValidator {
+
+    /** Authentication attribute containing SAML AuthenticationMethod attribute value. */
+    public static final String AUTH_METHOD_ATTRIBUTE = "samlAuthenticationStatement::authMethod";
+
+    /** SAML 1.1 request template. */
+    private static final String SAML_REQUEST_TEMPLATE;
+
+    /** SAML 1.1. namespace context. */
+    private static final NamespaceContext NS_CONTEXT = new MapNamespaceContext(
+            "soap->http://schemas.xmlsoap.org/soap/envelope/",
+            "sa->urn:oasis:names:tc:SAML:1.0:assertion",
+            "sp->urn:oasis:names:tc:SAML:1.0:protocol");
+
+    /** XPath expression to extract Assertion validity start date. */
+    private static final ThreadLocalXPathExpression XPATH_ASSERTION_DATE_START =
```

Comment:
These components are not thread safe, and their construction involves relatively expensive parsing of XPath expressions, which I wanted to avoid on every parse operation. Again, this may be premature optimization.